### PR TITLE
Remove syntax `: typ` in `Obligation` command (it was ignored)

### PR DESCRIPTION
**Pull request by SkySkimmer**

### CODE REVIEW

**Comment by Zimmi48** [comment]
Syntactically speaking, it looks to me like a backward-incompatible change with no clear migration path: despite being not used, the `: typ` syntax was compulsory to use the `with` clause. Am I mistaken?

I can imagine that this will not create any issue because, perhaps, nobody is using the `with` clause.

I guess the migration pathway we can recommend is to use `Next Obligation` instead?

File: plugins/ltac/g_obligations.mlg [hunk 1]
```diff
@@ -77,14 +76,10 @@ let classify_obbl _ = Vernacextend.(VtStartProof (Doesn'tGuaranteeOpacity,[]))
 }
 
 VERNAC COMMAND EXTEND Obligations CLASSIFIED BY { classify_obbl } STATE declare_program
-| [ "Obligation" natural(num) "of" identref(name) ":" lglob(t) withtac(tac) ] ->
-    { obligation (num, Some name.CAst.v, Some t) tac }
 | [ "Obligation" natural(num) "of" identref(name) withtac(tac) ] ->
```
Comment (SkySkimmer) [inline plugins/ltac/g_obligations.mlg:79]: this grammar rule has not changed and gives access to `with tac`

The doc on master makes it look like the type is need to use `with tac` but that seems to be a doc-only bug

Comment (Zimmi48) [reply plugins/ltac/g_obligations.mlg:79]: Ah, good! Then OK to merge this without needing a deprecation phase.

**Review by Zimmi48 — APPROVED** [review]

**Review by jfehrle — APPROVED** [review]

**Comment by SkySkimmer** [comment]
@coqbot run full ci

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
@coqbot merge now

**Pull request merged**